lsp/nls/src/requests/completion.rs

```diff
@@ -217,17 +217,22 @@ fn find_fields_from_contracts(
 ) -> Vec<IdentWithType> {
     annot
         .iter()
-        .flat_map(|contract| match &contract.types {
-            Types(TypeF::Record(row)) => find_fields_from_type(row, path, info),
-            Types(TypeF::Dict(ty)) => {
-                if let (Types(TypeF::Flat(term)), Some(_)) = (&**ty, path.pop()) {
-                    find_fields_from_term(term, path, info)
-                } else {
-                    Vec::new()
-                }
+        .flat_map(|contract| {
+            let mut path_copy = path.clone();
+            match &contract.types {
+                Types(TypeF::Record(row)) => find_fields_from_type(row, &mut path_copy, info),
+                Types(TypeF::Dict(ty)) => match (&**ty, path_copy.pop()) {
+                    (Types(TypeF::Flat(term)), Some(_)) => {
+                        find_fields_from_term(term, &mut path_copy, info)
+                    }
+                    (Types(TypeF::Record(rrows)), Some(_)) => {
+                        find_fields_from_type(rrows, &mut path_copy, info)
+                    }
```

Comment:
It seems this code is more or less similar to the outer code, but just nested (check if something is flat or a record). Should we put that in a separate subfunction `find_field_from_type` (and rename `find_field_from_type` to `find_field_from_rrows`? And should we just call it recursively, so that a nested dictionary type `{_: {_: {_: {bar}}}}` gives completion for `baz.rlog.blirg[.]` ?

Reply:
Yesss. Thank you!